### ISSUE TITLE
copy text from pc to phone

### ISSUE THREAD

**Issue by Karan-Sahota** [Genymobile/scrcpy#3611]
- I have checked that a similar [feature request does not already exist.

 i 've used the copy past function to 
 copy and paste text from phone to pc
  but  not the other way 

**  i realy don't know if its possible or
   not but  it would be great if  we could 
    also copy text  from pc to phone 
   that would  save a lot of time of 
    moving text here and there

**Comment by rom1v** [comment]
In theory, it works by default. However, on some devices (especially Samsung devices it seems), it does not work because setting the device clipboard does not work.

In that case, you can pass `--legacy-paste` to inject characters one by one.

See:
 - https://github.com/Genymobile/scrcpy#copy-paste
 - d5f059c7cbef7d13889086670716d9f33f805dc4

**Comment by AndroidDeveloperLB** [comment]
@rom1v For me, it always worked fine on my Pixel 6 device.
However, on the recent Android 13 update, I've noticed that the other way doesn't work, meaning from ScrCpy to other windows. I can copy from other windows and paste on ScrCpy, but I can't copy from ScrCpy and paste on other windows.

Can you please check?

My Android build:
TQ1A.221205.011

**Comment by chilicoke** [comment]
Having the same issue, used to copy and paste from my PC all the time but it no longer works on Pixel 6 Android 13.

**Comment by AndroidDeveloperLB** [comment]
@chilicoke Do you have perhaps the same build as me? I wonder if on the beta it happens, too

**Comment by chilicoke** [comment]
I'm on the same TQ1A.221205.011 build, never been on Beta on this phone.

**Comment by drothenberger** [comment]
@chilicoke Check out https://github.com/Genymobile/scrcpy/issues/3497#issuecomment-1246780900. It fixes the issue for me with the same build.

**Comment by chilicoke** [comment]
> @chilicoke Check out [#3497 (comment)](https://github.com/Genymobile/scrcpy/issues/3497#issuecomment-1246780900). It fixes the issue for me with the same build.

That fixed it! Absolutely perfect, thank you so much!

**Comment by AndroidDeveloperLB** [comment]
@drothenberger This worked for me. How did you find it? Why isn't it on the releases ?

**Comment by drothenberger** [comment]
I just posted a link to a comment by the developer in another similar issue that contains a fix. I assume the dev will include it in the next release.

**Comment by AndroidDeveloperLB** [comment]
@drothenberger Thank you!
I was beginning to think that ScrCpy won't work well anymore on Android, because of yet more restrictions that Google adds to it...

**Comment by ronvdlg** [comment]
Unfortunately I was never able to correctly use the clipboard (Windows 10) on an Android 10 using a Samsung Galaxy J4. When I inject characters one by one (legacy paste or Shift Alt V), it sends all break lines like "Enter / Ctrl Enter", sending messages in chats. The only alternative I found was to use Microsoft's SwiftKey app to sync my Windows clipboard with my Android clipboard.

**Comment by AndroidDeveloperLB** [comment]
@ronvdlg I thought that clipboard handling got very restricted on Android. Still possible on Samsung?

**Comment by AndroidDeveloperLB** [comment]
@rom1v This issue has come back again, on v3.2:
I can copy on PC and paste on the mirrored device fine, but not the opposite. 

What should I do? Please check it out...
:(

**Comment by rom1v** [comment]
> This issue has come back again, on v3.2

Please make sure this is related to 3.2: test with an older version (3.1, 3.0).

It it's the same, the problem is not 3.2, but related to an Android upgrade.

**Comment by AndroidDeveloperLB** [comment]
@rom1v It is. I wrote that it came back. It didn't exist on previous version.

I even talked with you somewhere about previous version (3.1), that it failed to handle orientation change, and you provided me with a special file to fix it:
https://github.com/Genymobile/scrcpy/issues/5930#issuecomment-2727377010

Weird thing is that now after I've closed ScrCpy and restarted it (v3.2, the newest), I don't see the issue anymore.
How weird...
Maybe was a temporary issue.

What should I do if the issue comes back?
Any way to capture logs and send you, even though I use this usually:

```
C:\android\scrcpy\scrcpy-noconsole.vbs -S --always-on-top --stay-awake --no-audio --keyboard=uhid  --no-mouse-hover
```

?

**Comment by rom1v** [comment]
Yes, the "legacy paste" method can only inject ASCII chars. That's why a new method is better… when it works on the device.
